### PR TITLE
AFRPAY-12135 update api version to use v2

### PR DESCRIPTION
**Pull request by ashrafhofney**

### CODE REVIEW

File: inc/WC_JumiaPay_Purchase.php [hunk 1]
```diff
@@ -83,58 +71,60 @@ private function getBasket() {
     }
 
     public function generateData() {
+        $merchantReferenceId = $this->generateMerchantReference();
 
         return [
-            "shopConfig" => $this->shopConfig,
+            "description" => "Payment for order #" . $merchantReferenceId,
```
Comment (cacevedojumee) [inline inc/WC_JumiaPay_Purchase.php:76]: please do not forget to sanitize and validate max length here too.

File: inc/WC_JumiaPay_Purchase.php [hunk 2]
```diff
@@ -83,58 +71,61 @@ private function getBasket() {
...
             ],
-            "priceCurrency"=> $this->currency,
-            "priceAmount"=> sanitize_text_field($this->order->get_total()),
-            "purchaseReturnUrl"=>  esc_url_raw($this->baseUrl."/wc-api/payment_return/?orderid=".$this->order->get_id()),
-            "purchaseCallbackUrl"=> esc_url_raw($this->baseUrl."/wc-api/payment_callback/?orderid=".$this->order->get_id()),
             "shippingAddress"=> [
-                "addressLine1"=> sanitize_text_field($this->order->get_shipping_address_1()),
-                "addressLine2"=> sanitize_text_field($this->order->get_shipping_address_2()),
-                "city"=> sanitize_text_field($this->order->get_shipping_city()),
+                "addressPrimary"=> sanitize_text_field($this->order->get_shipping_address_1()),
```
Comment (cacevedojumee) [inline inc/WC_JumiaPay_Purchase.php:102]: Please validate up to 512 chars as in the documentation. Apply to billing Address too.

File: inc/WC_JumiaPay_Purchase.php [hunk 3]
```diff
@@ -83,58 +71,61 @@ private function getBasket() {
...
+                "addressSecondary"=> sanitize_text_field($this->order->get_shipping_address_2()),
```
Comment (cacevedojumee) [inline inc/WC_JumiaPay_Purchase.php:103]: Please validate up to 512 chars as in the documentation. Apply to billing Address too.

File: inc/WC_JumiaPay_Purchase.php [hunk 4]
```diff
@@ -83,58 +71,61 @@ private function getBasket() {
...
+                "city"=> sanitize_text_field(substr($this->order->get_shipping_city(), 0, 50)),
                 "district"=> sanitize_text_field($this->order->get_shipping_state()),
```
Comment (cacevedojumee) [inline inc/WC_JumiaPay_Purchase.php:105]: Please validate up to 50  chars as in the documentation..Apply to billing Address too.

File: inc/WC_JumiaPay_Purchase.php [hunk 5]
```diff
@@ -83,58 +71,61 @@ private function getBasket() {
...
                 "province"=> sanitize_text_field($this->order->get_shipping_state()),
```
Comment (cacevedojumee) [inline inc/WC_JumiaPay_Purchase.php:106]: Please validate up to 50  chars as in the documentation. Apply to billing Address too.

**Review by cacevedojumee — CHANGES_REQUESTED** [review]
I added a few quick comments. The rest LGTM

File: inc/WC_JumiaPay_Purchase.php (hunk 5 above)
Comment (ashrafhofney) [reply inc/WC_JumiaPay_Purchase.php]: done

File: inc/WC_JumiaPay_Purchase.php (hunk 4 above)
Comment (ashrafhofney) [reply inc/WC_JumiaPay_Purchase.php]: done

File: inc/WC_JumiaPay_Purchase.php (hunk 3 above)
Comment (ashrafhofney) [reply inc/WC_JumiaPay_Purchase.php]: done

File: inc/WC_JumiaPay_Purchase.php (hunk 2 above)
Comment (ashrafhofney) [reply inc/WC_JumiaPay_Purchase.php]: done

File: inc/WC_JumiaPay_Purchase.php (hunk 1 above)
Comment (ashrafhofney) [reply inc/WC_JumiaPay_Purchase.php]: done

**Review by cacevedojumee — DISMISSED** [review]

**Review by jmsakr — APPROVED** [review]

**Review by cacevedojumee — APPROVED** [review]
LGTM

**Review by jmsakr — COMMENTED** [review]
are these changes covered by unit tests?

**Review by jmsakr — APPROVED** [review]

**Pull request merged**